hibernate-cache/src/main/java/org/infinispan/hibernate/cache/access/UnorderedDistributionInterceptor.java

```diff
@@ -64,43 +61,52 @@ public Object visitPutKeyValueCommand(InvocationContext ctx, PutKeyValueCommand
 				commandTopologyId + ", got " + currentTopologyId);
 		}
 
-		ConsistentHash writeCH = distributionManager.getWriteConsistentHash();
-		List<Address> owners = null;
-		if (writeCH.isReplicated()) {
+		LocalizedCacheTopology cacheTopology = distributionManager.getCacheTopology();
+		if (isReplicated) {
 			// local result is always ignored
-         List<Address> finalOwners = owners;
-         return invokeNextAndHandle( ctx, command, (rCtx, rCommand, rv, throwable) ->
-               invokeRemotelyAsync(finalOwners, rCtx, (WriteCommand) rCommand));
+			return invokeNextAndHandle(ctx, command, (rCtx, rCommand, rv, throwable) ->
+					invokeRemotelyAsync(null, rCtx, (WriteCommand) rCommand));
 		}
 		else {
-			owners = writeCH.locateOwners(command.getKey());
+			List<Address> owners = cacheTopology.getDistribution(command.getKey()).writeOwners();
 			if (owners.contains(rpcManager.getAddress())) {
-            List<Address> finalOwners = owners;
-            return invokeNextAndHandle( ctx, command, (rCtx, rCommand, rv, throwable) ->
-                  invokeRemotelyAsync(finalOwners, rCtx, (WriteCommand) rCommand));
+				return invokeNextAndHandle( ctx, command, (rCtx, rCommand, rv, throwable) ->
+                  invokeRemotelyAsync(owners, rCtx, (WriteCommand) rCommand));
```

Comment:
I would prefer keeping these files with the original formatting. If we do decide to change the formatting, I'd prefer a separate commit changing it everywhere.